### PR TITLE
multi: Improve proposals diff and versions

### LINKED ISSUE #2781: [plugin-architecture] Handle proposals diff and versions properly
Our currently implemented diff modal from proposals versions has a temporary behavior, and should provide more info about proposals diff.

I'd go for a better designed version of the current production site ([reference here](https://proposals.decred.org/record/e2d7b7d)):

Screen Shot 2022-06-03 at 4 55 46 PM

Also, records versions are being erased once you fetch a new version. **This should be also fixed!**

### PR DESCRIPTION
**Pull request by victorgcramos**
Closes #2781 

Depends on https://github.com/decred/pi-ui/pull/455

This PR fixes some issues on proposals versions by keeping different versions
of some record on our store. Previously we were replacing the values regardless
the record's version.

It also improves the proposals diff modal styles and UX, as well as allowing inline
images rendering.

### Preview

|  Before | After  | 
|---|---|
|  Screen Shot 2022-06-15 at 4 25 03 PM | Screen Shot 2022-06-24 at 4 30 01 PM  |
| Screen Shot 2022-06-24 at 4 34 29 PM |  Screen Shot 2022-06-24 at 4 28 38 PM |

### CODE REVIEW

**Comment by victorgcramos** [comment]
### Updated Preview:

Screen Shot 2022-06-16 at 6 23 53 PM

File: plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  width: 100%;
+  padding-bottom: 2rem;
+  border-bottom: 0.1px solid var(--separator-color);
+}
+
+.modalWrapper {
+  width: 100%;
+}
+
+.versionDownloads > * > * > * {
```
Comment (tiagoalvesdulce) [inline plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css:66]: is there any other way to write this selector?

File: plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+.removed {
+  border: 0.2rem solid var(--color-orange);
+}
+
+.subtitle {
+  color: var(--text-secondary-color) !important;
+}
+
+.modalContent {
+  min-height: 40vh;
```
Comment (tiagoalvesdulce) [inline plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css:49]: I think the modal can be a bit bigger. Maybe `50vh`?

File: plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+.diffWrapper > * > *:nth-child(2) {
+  margin: 0;
+}
+
+.diffHeader {
+  position: sticky;
+  background-color: var(--card-background);
+  padding: 2rem 0;
+  border-bottom: 0.1rem solid var(--separator-color);
+  top: -3rem;
```
Comment (tiagoalvesdulce) [inline plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css:21]: Can you elaborate on why we need this negative value?

File: plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+.versionPickerWrapper {
+  align-items: center;
+  display: flex;
+}
+
+@media screen and (max-width: 560px) {
+  .diffHeader {
+    top: -5rem;
```
Comment (tiagoalvesdulce) [inline plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css:87]: Can you elaborate?

File: plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+.added {
+  border: 0.2rem solid var(--color-green);
+}
+
+.removed {
+  border: 0.2rem solid var(--color-orange);
+}
+
+.subtitle {
+  color: var(--text-secondary-color) !important;
```
Comment (tiagoalvesdulce) [inline plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css:45]: this shouldn't need `!important`. If it does need, it's an issue with pi-ui.

**Review by tiagoalvesdulce — CHANGES_REQUESTED** [review]
In general, I would expect the CSS to be more organized. Using multi-level selectors like `.class > * > * > *` is very hard to read and maintain.

Also, if pi-ui CSS is always taking precedence over politeiagui CSS. We have to fix it there now. Before we pollute our CSS with unnecessary stuff.

**Comment by tiagoalvesdulce** [comment]
I have a proposal that two different versions look the same:

https://user-images.githubusercontent.com/13955303/175946513-da1da8e5-8112-498b-84cc-86b8b2c598b8.mp4

File: plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css (hunk 1 above)
Comment (victorgcramos) [reply plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css:66]: That's a pi-ui issue. Will fix that there!

Comment (victorgcramos) [reply plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css:66]: https://github.com/decred/pi-ui/issues/453 reference

Comment (victorgcramos) [reply plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css]: Pushed a fix on https://github.com/decred/pi-ui/pull/455

File: plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css (hunk 3 above)
Comment (victorgcramos) [reply plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css:22]: This is required to stick the element on top of the modal.

File: plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css (hunk 4 above)
Comment (victorgcramos) [reply plugins-structure/apps/politeia/src/components/Proposal/ModalProposalDiff.module.css:89]: > This is required to stick the element on top of the modal without extra weird spaces.

same here, but mobile screens have larger spaces

**Comment by victorgcramos** [comment]
About that unnoticed diff, it's due to markdown formatting behavior. 

After using the same payload as @tiagoalvesdulce, I realized that the difference between the two versions are from line breaks, which are not recognized by the Markdown Renderer. However, the diff is still relevant, so we decided to also add a **raw diff preview** to enhance the changes comparison UX.

**Review by tiagoalvesdulce — CHANGES_REQUESTED** [review]
Looks good. I like the toggle for markdown and raw diff.

The raw diff is overflowing for medium screen sizes. Also, the `version` dropdown are too small. See: 

Screen Shot 2022-06-29 at 15 57 57

**Comment by victorgcramos** [comment]
New view for medium screen sizes:

Screen Shot 2022-06-29 at 5 18 14 PM

**Pull request merged**